state/allwatcher.go

```diff
@@ -542,25 +544,12 @@ func (app *backingApplication) updated(st *State, store *multiwatcherStore, id s
 		if err != nil {
 			return errors.Annotatef(err, "reading application status for key %s", key)
 		}
-		if err == nil {
```

Comment:
This error is always non-nil. It is checked above and returns for nil, so none of this block is ever hit.

This is years old code. The bug referenced in the comment (https://bugs.launchpad.net/juju-core/+bug/1451283) dates from Juju version 1.

Reply:
Nice cleanup.